akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -130,6 +147,31 @@ private[akka] object Running {
         }
 
         source.runWith(Sink.ignore)(SystemMaterializer(system).materializer)
+
+        // FIXME support from journal to fast forward
+        // (how can we support/detect both this and journal ffwd in a backwards compatible way and fallback)
+        state.copy(
+          replicationControl =
+            state.replicationControl.updated(replicaId, new ReplicationStreamControl {
+              override def fastForward(sequenceNumber: Long): Unit = {
+                // (logging is safe here since invoked on message receive
+                OptionVal(controlRef.get) match {
```

Comment:
so we have this dance as we don't block the starting of the ESB until the replication stream has materialized? Should we?

Reply:
It is not because of that but because it is wrapped in a restart flow, which can fail and be replaced and we need to control the latest restarted query.

Reply:
Ah of course. 👍